### PR TITLE
[azure] fix sa creation failure

### PR DESCRIPTION
**Pull request by jinyunma**
[OCPQE-18458](https://issues.redhat.com/browse/OCPQE-18458)  Fail to create storage account in step azure-provision-bastionhost

Failed job: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.14-arm64-nightly-4.14-upgrade-from-stable-4.14-azure-ipi-fullyprivate-proxy-f28/1736624589921849344

Error:
```
Create a Storage Account for bastion vhd
Running Command: az storage account create -g ci-op-wncf4nvv-921c3-rg --name wncf4nvv921c3basa --kind Storage --sku Standard_LRS
WARNING: The public access to all blobs or containers in the storage account will be disallowed by default in the future, which means default value for --allow-blob-public-access is still null but will be equivalent to false.
 ERROR: (StorageAccountOperationInProgress) An operation is currently performing on this storage account that requires exclusive access.
Code: StorageAccountOperationInProgress
Message: An operation is currently performing on this storage account that requires exclusive access.
```
From [analysis](https://stackoverflow.com/questions/48640970/an-operation-is-currently-performing-on-this-storage-account-that-requires-exclu), seems that error occurs when creating parallel resources at the same time, adding retry to see if issue can be solved.

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-azure-ipi-fullyprivate-proxy-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-azure-ipi-fullyprivate-proxy-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-azure-ipi-fullyprivate-proxy-f14

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei to review
could not verify the update in the PR, might need to merge it and check in CI if issue happens again.

File: ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh [hunk 1]
```diff
@@ -117,7 +117,27 @@ echo "Create a Storage Account for bastion vhd"
...
+            echo "storage account ${sa_name} creation is completed."
+            break
+        fi
+        try=$(( try + 1 ))
+    done
+    if (( try == max_retries )); then
+        echo "ERROR: fail to create storage account ${sa_name}."
+        exit 1
+    fi
+fi
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh:140]: How about introduce a function for retires:
```
function run_command_with_retries()
{   
  local try=0 cmd="$1" retries="$2" ret
  [[ -z ${retries} ]] && max="20" || max=${retries}
  echo "Trying ${max} times max to run '${cmd}'"

  eval "${cmd}"; ret=$?
  while [ X"${ret}" != X"0" ] && [ ${try} -lt ${max} ]; do
    echo "'${cmd}' did not return success, waiting 60 sec....."
    sleep 60
    (( try++ ))
    eval "${cmd}"; ret=$?
  done
  if [ ${try} -eq ${max} ]; then
    echo "Never succeed or Timeout"
    return 4
  fi
  echo "Succeed"
  return 0
}
```

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-azure-ipi-fullyprivate-proxy-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-azure-ipi-fullyprivate-proxy-f14

File: ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh [hunk 2]
```diff
@@ -117,8 +138,8 @@ echo "Create a Storage Account for bastion vhd"
 # 'account_name' must have length less than 24, so hardcode the basion sa name
 sa_name_prefix=$(echo "${NAMESPACE}" | sed "s/ci-op-//" | sed 's/[-_]//g')
 sa_name="${sa_name_prefix}${UNIQUE_HASH}basa"
-run_command "az storage account create -g ${bastion_rg} --name ${sa_name} --kind Storage --sku Standard_LRS" &&
-account_key=$(az storage account keys list -g ${bastion_rg} --account-name ${sa_name} --query "[0].value" -o tsv) || exit 3
+run_command_with_retries "az storage account create -g ${bastion_rg} --name ${sa_name} --kind Storage --sku Standard_LRS"
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh:141]: the default max retries of `run_command_with_retries` is 20, is that a bit agressive?

Comment (jinyunma) [reply ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh]: changed to retry 5 times for this command, the function is verified with default max retires in [job](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/46954/rehearse-46954-periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-azure-ipi-fullyprivate-proxy-f14/1737374560879120384/artifacts/azure-ipi-fullyprivate-proxy-f14/azure-provision-bastionhost/build-log.txt) and passed.

**Pull request merged**